### ISSUE TITLE
Interpretability vs Generality

### ISSUE THREAD

**Issue by xloem** [karttikeya/minREV#1]
Hi,

Do you have any resources on the value of reducing generality to increase interpretability?

As a software engineer, I find general implementations to be more interpretable, assuming they use clear self commenting practices and such. It’s been frustrating to engage the ML ecosystem which uses so little generality. But it’s exciting to see your README where you describe this as something that is counterposed with interpretability, which explains things some.

Where does the value stem from? What is the logic? Is there existing literature on this?

**Comment by karttikeya** [comment]
Hi, 

Thanks for your comment. While I agree that the desirable state is both interpretable and general, here I have made the choice to choose interpretable over general should such a conflict arise. A good example is perhaps to contrast this with the pyslowfast implementation of reversible ViT.  PySF provides a more general framework for several video understanding tasks such as classificiation/detection, many many backbones (both convolutional, transformer based and reversible), several learning setups (supervised, semi-supervised, constrastive, MAE etc) and more. As a result, the codebase is quite big and not ideal to hack on. Here I've stripped away most of the complexity to the bare reversible minimum which allows easy understanding and interpretability, in my opinion. Another great example is [minGPT repo](https://github.com/karpathy/minGPT). 

Of course, more general solutions that still keep these values are welcome PRs.
